src/common/meta/src/key/table_info.rs

```diff
@@ -52,6 +55,58 @@ impl TableInfoManager {
         Self { kv_backend }
     }
 
+    pub async fn get_old(&self, table_name: &TableName) -> Result<Option<TableInfoValue>> {
+        let table_global_key = TableGlobalKey {
+            catalog_name: table_name.catalog_name.clone(),
+            schema_name: table_name.schema_name.clone(),
+            table_name: table_name.table_name.clone(),
+        };
+        self.kv_backend
+            .get(table_global_key.to_string().as_bytes())
+            .await?
+            .map(|kv| TableGlobalValue::from_bytes(kv.value))
+            .transpose()
+            .map(|v| {
+                v.map(|v| TableInfoValue {
+                    table_info: v.table_info,
+                    version: 0,
+                })
+            })
+            .context(InvalidCatalogValueSnafu)
+    }
+
+    pub async fn put_old(&self, table_info: RawTableInfo) -> Result<()> {
+        let key = TableGlobalKey {
+            catalog_name: table_info.catalog_name.clone(),
+            schema_name: table_info.schema_name.clone(),
+            table_name: table_info.name.clone(),
+        }
+        .to_string();
+        let raw_key = key.as_bytes();
+
+        let regions_id_map = self
+            .kv_backend
+            .get(raw_key)
+            .await?
+            .map(|kv| TableGlobalValue::from_bytes(kv.value()))
+            .transpose()
+            .context(InvalidCatalogValueSnafu)?
+            .map(|v| v.regions_id_map)
+            .unwrap_or_default();
+
+        let raw_value = TableGlobalValue {
+            node_id: 0,
+            regions_id_map,
+            table_info,
+        }
+        .as_bytes()
+        .context(InvalidCatalogValueSnafu)?;
+
+        let req = PutRequest::new().with_key(raw_key).with_value(raw_value);
+        self.kv_backend.put(req).await?;
```

Comment:
Is there any issue with concurrent updates when using get and put together, considering that they do not have atomic execution?

Reply:
It's possible, but it's a temporary solution, will soon use the new interface.